File: Data Structures & Algorithms/is-anagram/submission-5.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(a, t) {
        if(a.length!= t.length) return false
        let s = new Map()

        for(var char of a){
            let count = s.get(char) || 0
            s.set(char, count+1)
        }

        for(var char of t){
            if(!s.has(char)){
                return false
            }
            let count = s.get(char)
            if(count == 0 ){
                return false
            }
            s.set(char, count-1)
        }
        return true
    }

}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        res = []
        for i in nums:
            count[i] = count.get(i,0) + 1
        
        bucket = {x:[] for x in range(len(nums)+1)}

        for i in count:
             bucket[count[i]].append(i)
        
        for i in range(len(nums),0,-1):
            for j in range(len(bucket[i])):
                res.append(bucket[i][j])
                if(len(res)==k):
                    return res
        
        return res






